### PR TITLE
Support back

### LINKED ISSUE #136: Support back button on "getting started" page
We need to add support for the back button the Getting Stared workflow page. Currently you have to use the "reset guide" button to use the page again (or refresh).

### PR DESCRIPTION
**Pull request by arichiv-usds**
closes #136 

Thanks to @fulldecent for the implementation tips!

I implemented this using hash-based history storage, separating entries with a `!`. There's a good amount of plumbing, but in the end, the rendering code got easier to read (we can simply re-draw each time).

### CODE REVIEW

File: assets/js/main.js [hunk 1]
```diff
@@ -82,46 +82,77 @@
     }
 
     var nodes = find('.tree-node', tree);
-    nodes[0].classList.add('active');
+    var hashHistory = [];
 
+    // Pull tree history from hash
+    var updateHashHistory = function() {
```
Comment (jordangov) [inline assets/js/main.js]: I'd like to be consistent, and in other places we're using function literals (line 57) versus function expressions (as here). Can we pick one?

**Comment by jordangov** [comment]
This is good overall. I want to double check it on my local machine (or maybe @jgarber623-gov can). After that (and the one comment), let's merge.

**Comment by fulldecent** [comment]
Looks great! Walkthrough:

File: assets/js/main.js (hunk 1 above)
Comment (arichiv-usds) [reply assets/js/main.js]: Can I take a JS cleanup pass in a future PR (after this is in)?

**Pull request merged**